### DISCUSSION TITLE
how to copy the request after extract the request?

### DISCUSSION THREAD

**Discussion by godcheese** [tokio-rs/axum#2109] [Q&A]
hi guys.
how to copy the request after extract the request?

**Comment by davidpdrsn** [comment]
You can't copy the request since the body is an async stream and request extensions don't support cloning. See https://github.com/tokio-rs/axum/blob/main/examples/consume-body-in-extractor-or-middleware/src/main.rs for a way to deal with that if you need to consume the request body in middleware.
